src/Facility.CodeGen.JavaScript/JavaScriptGenerator.cs

```diff
@@ -316,24 +502,52 @@ private string RenderUriComponent(ServiceFieldInfo field, ServiceInfo service)
 
 			switch (fieldTypeKind)
 			{
-			case ServiceTypeKind.Enum:
-				return $"request.{fieldName}";
-			case ServiceTypeKind.String:
-			case ServiceTypeKind.Bytes:
-				return $"encodeURIComponent(request.{fieldName})";
-			case ServiceTypeKind.Boolean:
-			case ServiceTypeKind.Int32:
-			case ServiceTypeKind.Int64:
-			case ServiceTypeKind.Decimal:
-				return $"request.{fieldName}.toString()";
-			case ServiceTypeKind.Double:
-				return $"encodeURIComponent(request.{fieldName}.toString())";
-			case ServiceTypeKind.Dto:
-			case ServiceTypeKind.Error:
-			case ServiceTypeKind.Object:
-				throw new NotSupportedException("Field type not supported on path/query: " + fieldTypeKind);
-			default:
-				throw new NotSupportedException("Unknown field type " + fieldTypeKind);
+				case ServiceTypeKind.Enum:
+					return $"request.{fieldName}";
+				case ServiceTypeKind.String:
+				case ServiceTypeKind.Bytes:
+					return $"encodeURIComponent(request.{fieldName})";
+				case ServiceTypeKind.Boolean:
+				case ServiceTypeKind.Int32:
+				case ServiceTypeKind.Int64:
+				case ServiceTypeKind.Decimal:
+					return $"request.{fieldName}.toString()";
+				case ServiceTypeKind.Double:
+					return $"encodeURIComponent(request.{fieldName}.toString())";
+				case ServiceTypeKind.Dto:
+				case ServiceTypeKind.Error:
+				case ServiceTypeKind.Object:
+					throw new NotSupportedException("Field type not supported on path/query: " + fieldTypeKind);
+				default:
+					throw new NotSupportedException("Unknown field type " + fieldTypeKind);
+			}
+		}
+
+		private string RenderJsConversion(ServiceFieldInfo field, ServiceInfo service, string value)
+		{
+			var fieldTypeKind = service.GetFieldType(field).Kind;
+			var fieldName = field.Name;
+
+			switch (fieldTypeKind)
+			{
+				case ServiceTypeKind.Enum:
+				case ServiceTypeKind.String:
+				case ServiceTypeKind.Bytes:
+					return value;
+				case ServiceTypeKind.Boolean:
+					return $"{value} === 'true'";
```

Comment:
I really need to document the syntax for query string parameter values.

Consider returning null or undefined if the value isn't true or false. That's what the C# implementation currently does, though I wonder if it should return an invalid request.

Also, C# uses a case-insensitive match. But I think I will allow implementations to be case-sensitive. Certainly JS/TS clients would be a lot harder to write if case-insensitivity were allowed everywhere.